src/util/test_util.rs

```diff
@@ -0,0 +1,13 @@
+//! Utility functions that are useful for tests.
+
+#[inline(always)]
+/// Compare two `f32`s, and assert equality.
+pub fn assert_f32_equal(a: f32, b: f32) {
+    assert!((a - b).abs() < std::f32::EPSILON);
+}
+
+#[inline(always)]
+/// Assert that an `f32` is zero.
+pub fn assert_f32_zero(a: f32) {
+    assert!(a.abs() < std::f32::EPSILON);
+}
```

Comment:
This function seems redundant (and has a misleading name), since comparing against 0.0 using the approximate comparison function would be at least as clear.